tests/search/test_search.py

```diff
@@ -1,106 +1,140 @@
+            company_name="Test Startup",
+            description="AI Startup in healthcare",
+            stage="seed",
+            founded_year=2020,
+            industry=self.industry,
+            location=self.location,
+            email="hello@test-startup.com",
+            team_size=5,
+        )
 
+        changed = False
+        for flag in ("is_active", "is_published", "is_verified", "published", "visible"):
+            if hasattr(self.startup, flag) and not getattr(self.startup, flag):
+                setattr(self.startup, flag, True)
+                changed = True
+        if changed:
+            self.startup.save()
 
+        # Project
+        self.project = Project.objects.create(
+            startup=self.startup,
+            category=self.category,
+            title="Health AI", 
+            description="AI project for healthcare",
+            status="active",   
+            funding_goal=Decimal("10000.00"),
+            current_funding=Decimal("0.00"),
+            email="project@test-startup.com",
+        )
+        changed = False
+        for flag in ("is_active", "is_published", "is_verified", "published", "visible"):
+            if hasattr(self.project, flag) and not getattr(self.project, flag):
+                setattr(self.project, flag, True)
+                changed = True
+        if changed:
+            self.project.save()
 
+    @staticmethod
+    def _make_hit(doc_id):
+        """ES hit supporting both hit.id and hit.meta.id access."""
+        hit = mock.Mock()
+        hit.id = str(doc_id)
+        hit.meta = mock.Mock()
+        hit.meta.id = str(doc_id)
+        return hit
 
+    def _chainable_search(self, search_obj, hits, qs=None):
+        """
+        Prepare a mocked search object compatible with the SUT:
+          - iteration directly over search (for hit in search)
+          - .query/.filter/.source/.extra return the same object
+          - slicing [:N] returns the same object
+          - .execute() also returns iterable(hits)
+          - (optional) .to_queryset() -> qs
+        """
+        for m in ("query", "filter", "source", "extra"):
+            getattr(search_obj, m).return_value = search_obj
+
+        search_obj.__getitem__.return_value = search_obj
+
+        search_obj.__iter__ = lambda s: iter(hits)
+
+        mock_resp = mock.Mock()
+        mock_resp.__iter__ = lambda s: iter(hits)
+        search_obj.execute.return_value = mock_resp
+
+        if qs is not None:
+            search_obj.to_queryset.return_value = qs
+
+    def test_startup_search_mocked(self):
+        with mock.patch.object(sv.StartupDocument, "search") as search_fn:
+            search_obj = search_fn.return_value
+            hit = self._make_hit(self.startup.id)
+            qs = Startup.objects.filter(pk=self.startup.pk)
+            self._chainable_search(search_obj, [hit], qs=qs)
+
+            url = reverse("startup-search")
+            resp = self.client.get(url, {"q": "health"}) 
```

Comment:
Hardcoded query "health" may make the test brittle if search logic changes. Use a variable for the search term for consistency.
```
query = "health"
resp = self.client.get(url, {"q": query})
```